bson/src/main/org/bson/FieldNameValidator.java

```diff
@@ -37,4 +37,18 @@
      * @return a non-null validator
      */
     FieldNameValidator getValidatorForField(String fieldName);
+
+    /**
+     * Start validation of a single document.
+     *
+     * @since 4.0
+     */
+    void start();
```

Comment:
As we are now in Java 8 - would it be better to use a default no-op method for start and end?

Reply:
+1

The advantage is that existing validators (our own and third-party) will continue to work without change, and only have to override the default methods if they really need to.

Reply:
@rozza resolving this as you've already LGTM'd